src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,801 @@
+
+  private meridiemEl: HTMLSpanElement;
+
+  private hourEl: HTMLSpanElement;
+
+  private minuteEl: HTMLSpanElement;
+
+  private secondEl: HTMLSpanElement;
+
+  private timeChanged = false;
+
+  // --------------------------------------------------------------------------
+  //
+  //  State
+  //
+  // --------------------------------------------------------------------------
+
+  /** The am/pm value */
+  @State() meridiem: Meridiem = null;
+
+  @State() displayHour: string = this.getDisplayHour();
+
+  @State() editingHourWhileFocused = false;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * @internal
+   */
+  @Event() calciteTimePickerBlur: EventEmitter<Time>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteTimePickerChange: EventEmitter<Time>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteTimePickerFocus: EventEmitter<Time>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("blur")
+  hostBlurHandler(): void {
+    this.calciteTimePickerBlur.emit();
+  }
+
+  @Listen("focus")
+  hostFocusHandler(): void {
+    this.calciteTimePickerFocus.emit();
+  }
+
+  @Listen("keydown")
+  keyDownHandler(event: KeyboardEvent): void {
+    switch (this.activeEl) {
+      case this.hourEl:
+        if (event.key === "ArrowRight") {
+          this.setFocus("minute");
+        }
+        break;
+      case this.minuteEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            this.setFocus("hour");
+            break;
+          case "ArrowRight":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else if (this.hourDisplayFormat === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.secondEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            this.setFocus("minute");
+            break;
+          case "ArrowRight":
+            if (this.hourDisplayFormat === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.meridiemEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else {
+              this.setFocus("minute");
+            }
+            break;
+        }
+        break;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(target: TimeComponents): Promise<void> {
+    this[`${target || "hour"}El`].focus();
```

Comment:
Should this have a check to make sure the node is there before focusing?

Reply:
that's what the `target || "hour"` should do.  It would fall back to the `hourEl` which should always be there.

Reply:
Guess I could add a sanity check there anyway.  Good idea.